crates/ruff_linter/src/rules/pylint/rules/stop_iteration_return.rs

```diff
@@ -0,0 +1,115 @@
+use ruff_macros::{ViolationMetadata, derive_message_formats};
+use ruff_python_ast as ast;
+use ruff_python_ast::visitor::{Visitor, walk_expr, walk_stmt};
+use ruff_text_size::Ranged;
+
+use crate::Violation;
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for explicit `raise StopIteration` in generator functions.
+///
+/// ## Why is this bad?
+/// Raising `StopIteration` in a generator function causes a `RuntimeError`
+/// at runtime (even in Python 3.7+, the earliest version we support). This
+/// breaks the abstraction between generators and iterators and will crash
+/// the program when the generator is iterated over.
```

Comment:
The first part here feels a bit redundant, and then I think the fact that it raises a runtime error is a bit more severe than breaking an abstraction. I think we could just say something like:


```suggestion
/// Raising `StopIteration` in a generator function causes a `RuntimeError`
/// when the generator is iterated over.
```